lib/string_decoder.js

```diff
@@ -262,20 +276,28 @@ function base64Text(buf, i) {
     this.lastChar[0] = buf[buf.length - 2];
     this.lastChar[1] = buf[buf.length - 1];
   }
-  return buf.toString('base64', i, buf.length - n);
+  return base64Slice(buf, i, buf.length - n);
 }
 
 
 function base64End(buf) {
   const r = (buf && buf.length ? this.write(buf) : '');
   if (this.lastNeed)
-    return r + this.lastChar.toString('base64', 0, 3 - this.lastNeed);
+    return r + this.lastChar.base64Slice(0, 3 - this.lastNeed);
```

Comment:
I think this can also be changed to `return r + base64Slice(this.lastChar, 0, 3 - this.lastNeed)`